.travis.yml

```diff
@@ -1,3 +1,16 @@
-script:
-- docker build --target all_tests -t test . && docker run --name todo-app-test --env TRELLO_URL --env TRELLO_KEY --env TRELLO_BOARD_ID --env TRELLO_TOKEN test
-- docker build --target prod -t prod .
+jobs:
+  include:
+    - stage: test
+      script:
+        - docker build --target all_tests -t test .
+        - docker run --name todo-app-test --env TRELLO_URL --env TRELLO_KEY --env TRELLO_BOARD_ID --env TRELLO_TOKEN test
+    - stage: deploy
+      script:
+        - DOCKER_REPO="$DOCKER_USERNAME/todo-app"
+        - docker build --target prod -t "$DOCKER_REPO:$TRAVIS_COMMIT" -t "$DOCKER_REPO:latest" .
+        - docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
```

Comment:
For our login step, we should really switch this to use the `--password-stdin` flag [as per the docker docs](https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin) (e.g. as below) to avoid the password ending up in shell logs/history
```
echo $PASSWORD | docker login -u $USERNAME --password-stdin
```